assets/scripts/segments/resize_drag.js

```diff
@@ -0,0 +1,97 @@
+import { setIgnoreStreetChanges } from '../streets/data_model'
+import { INFO_BUBBLE_TYPE_SEGMENT } from '../info_bubble/constants'
+import { infoBubble } from '../info_bubble/info_bubble'
+import {
+  RESIZE_TYPE_INITIAL,
+  RESIZE_TYPE_PRECISE_DRAGGING,
+  resizeSegment,
+  cancelFadeoutControls
+} from './resizing'
+import { segmentsChanged, getSegmentEl } from './view'
+import { TILE_SIZE } from './constants'
+import store from '../store'
+
+let originalWidth
+
+function getActiveSegmentWidth () {
+  const activeSegment = store.getState().ui.activeSegment
```

Comment:
I noticed that problem where while resizing it suddenly switches the segment that is being resized? I can't recreate it reliably but it seems to occur when I'm resizing pretty quickly.

Oh! This might actually be due to inside the `<Segment /`> component in `onSegmentMouseEnter` that if the mouse enter is suppressed I update the activeSegment to figure out which infoBubble to show next. Maybe?

Reply:
Hmm. I set this check up in the reducer itself. If `resizeDragState` is true, the `SET_ACTIVE_SEGMENT` action in the reducer will never update the active segment. At least, that's how it's supposed to work.

That way it doesn't matter who tries to call the `setActiveSegment()` action creator; they don't need to check the drag state themselves.

The question I now have is whether something might be accidentally setting `resizeDragState` to false....